apps/ensindexer/src/config/config.schema.ts

```diff
@@ -12,177 +12,179 @@ import {
 
-const parseDatabaseUrl = () =>
-  z.union(
-    [
-      z.string().refine((url) => {
-        try {
-          if (!url.startsWith("postgresql://") && !url.startsWith("postgres://")) {
-            return false;
-          }
-          const config = parseConnectionString(url);
-          return !!(config.host && config.port && config.database);
-        } catch {
+const RpcConfigsSchema = z.record(
+  z.string().transform(Number).pipe(chainIdSchema),
+  RpcConfigSchema,
+  {
+    error: "Chains configuration must be an object mapping numeric chain IDs to their configs.",
+  },
+);
+
+const DatabaseUrlSchema = z.union(
+  [
+    z.string().refine((url) => {
+      try {
+        if (!url.startsWith("postgresql://") && !url.startsWith("postgres://")) {
           return false;
         }
-      }),
-      z.undefined(),
-    ],
-    {
-      message:
-        "Invalid PostgreSQL connection string. Expected format: postgresql://username:password@host:port/database",
-    },
-  );
+        const config = parseConnectionString(url);
+        return !!(config.host && config.port && config.database);
+      } catch {
+        return false;
+      }
+    }),
+    z.undefined(),
+  ],
+  {
+    message:
+      "Invalid PostgreSQL connection string. Expected format: postgresql://username:password@host:port/database",
+  },
+);
+
+const derive_isSubgraphCompatible = (
+  config: Omit<ENSIndexerConfig, "isSubgraphCompatible">,
+): ENSIndexerConfig => {
+  const onlySubgraphPluginActivated =
+    config.plugins.length === 1 && config.plugins[0] === PluginName.Subgraph;
+  const indexingBehaviorIsSubgraphCompatible =
+    !config.healReverseAddresses && !config.indexResolverRecords;
+
+  return {
+    ...config,
+    isSubgraphCompatible: onlySubgraphPluginActivated && indexingBehaviorIsSubgraphCompatible,
+  };
+};
 
 const ENSIndexerConfigSchema = z
   .object({
-    ensDeploymentChain: parseEnsDeploymentChain(),
-    globalBlockrange: parseBlockrange(),
-    ensNodePublicUrl: parseEnsNodePublicUrl(),
-    ensAdminUrl: parseEnsAdminUrl(),
-    ponderDatabaseSchema: parsePonderDatabaseSchema(),
-    plugins: parsePlugins(),
-    healReverseAddresses: parseHealReverseAddresses(),
-    port: parsePort(),
-    ensRainbowEndpointUrl: parseEnsRainbowEndpointUrl(),
-    rpcConfigs: parseRpcConfigs(),
-    databaseUrl: parseDatabaseUrl(),
+    ensDeploymentChain: EnsDeploymentChainSchema,
+    globalBlockrange: BlockrangeSchema,
+    ensNodePublicUrl: EnsNodePublicUrlSchema,
+    ensAdminUrl: EnsAdminUrlSchema,
+    ponderDatabaseSchema: PonderDatabaseSchemaSchema,
+    plugins: PluginsSchema,
+    healReverseAddresses: HealReverseAddressesSchema,
+    indexResolverRecords: IndexResolverRecordsSchema,
+    port: PortSchema,
+    ensRainbowEndpointUrl: EnsRainbowEndpointUrlSchema,
+    rpcConfigs: RpcConfigsSchema,
+    databaseUrl: DatabaseUrlSchema,
   })
+  // inject ENSIndexerConfig.isSubgraphCompatible
+  .transform(derive_isSubgraphCompatible)
```

Comment:
Thinking about deriving the `ensDeployment` config prop. Right now, whenever we need to access the current ENS Deployment object, we'd import `getENSDeployment` function and pass the `config.ensDeploymentChain` as a param:
![image](https://github.com/user-attachments/assets/0f92a368-433c-459b-99e3-6d4117434bb0)

I propose we derive `ensDeployment` property and include it on `ENSIndexerSchema`

```suggestion
  .transform(derive_isSubgraphCompatible)
  .transform(derive_ensDeployment)
```

Where `derive_ensDeployment` is defined as the following:
```ts
const derive_ensDeployment = (
  config: Omit<ENSIndexerConfig, "ensDeployment">,
): ENSIndexerConfig {
  return {
    ...config,
    ensDeployment: getENSDeployment(config.ensDeploymentChain),
  };

Reply:
If we feel the idea is OK, but it needs a follow up PR, I'm happy to create a GH issue ;) 

Also, thinking of deriving active plugins list in a similar fashion, and later using this list in `ponder.config.ts` to actually activate relevant plugins.

Reply:
actually, splitting this into a separate issue